Library/Homebrew/utils/ruby.sh

```diff
@@ -0,0 +1,34 @@
+setup-ruby-path() {
+  if [[ -z "$HOMEBREW_DEVELOPER" ]]
+  then
+    unset HOMEBREW_RUBY_PATH
+  fi
+
+  if [[ -z "$HOMEBREW_RUBY_PATH" && "$HOMEBREW_COMMAND" != "install-vendor" ]]
+  then
+    HOMEBREW_VENDOR_RUBY_PATH="$HOMEBREW_LIBRARY/Homebrew/vendor/ruby/opt/bin/ruby"
+    if [[ -x "$HOMEBREW_VENDOR_RUBY_PATH" ]]
+    then
+      HOMEBREW_RUBY_PATH="$HOMEBREW_VENDOR_RUBY_PATH"
+    else
+      if [[ -n "$HOMEBREW_OSX" ]]
+      then
+        HOMEBREW_RUBY_PATH="/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby"
+      else
+        HOMEBREW_RUBY_PATH="$(which ruby)"
+      fi
+
+      if [[ -z "$HOMEBREW_RUBY_PATH" || "$("$HOMEBREW_RUBY_PATH" -e "puts RUBY_VERSION.split('.').first")" != "2" ]]
```

Comment:
@xu-cheng Just occurred to me: might also be nice to have an opt-in variable to use the vendored Ruby even if `HOMEBREW_RUBY_PATH` is correct. I could see this being useful for CI to make sure we're testing our vendored Ruby there on e.g. Mavericks.